### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by muhenge**
## In this project we:
- Created forms in the view(HTML forms, `form_tag`, `form_for`)
- Worked with those forms to **Create**, **Update**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (maelfosso) [inline app/controllers/users_controller.rb:7]: Remove comments

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    # @user.update(username: params[:username], email: params[:email], password: params[:password])
```
Comment (maelfosso) [inline app/controllers/users_controller.rb:23]: Remove comment

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    @user.update(user_params)
```
Comment (maelfosso) [inline app/controllers/users_controller.rb:24]: That snippet is not working
Update it like the **save** method
```rb
if @user.update
...
else
...
end
```

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<% if false %>
+ <form action="/users" method="post" accept-charset="UTF-8">
```
Comment (maelfosso) [inline app/views/users/new.html.erb:2]: It's preferable to use `users_path`

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  <label for="email">email:</label>
+  <input type="email" id="email" name="user[email]"><br><br>
+    <label for="password">Password:</label>
+  <input type="password" id="password" name="user[password]"><br><br>
+  <input type="submit" value="Submit">
+</form>
+<% end %>
+
+<% if false %>
+<%= form_tag('/users') do %>
```
Comment (maelfosso) [inline app/views/users/new.html.erb:15]: It's preferable to use `users_path`

**Comment by uryelah** [comment]
Hi **Herve Ngenzi & Josia Rakotonarivo**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with the forms and controller.
On an optional note, consider also showing up error messages if the user enters a repeated email.

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by uryelah** [comment]
Hi **Hillary Odhiambo**, I'm Sarah your reviewer for this PR 👩‍🚀 

You did a great job with your tests! Next time just remember to include the classes/methods you are testing in the same branch you write the tests in.

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**